src/v/cluster/types.cc

```diff
@@ -293,24 +318,188 @@ std::ostream& operator<<(
 } // namespace cluster
 
 namespace reflection {
+
+// These enum encoding/decoding functions should be move into the different
+// translation unit eventually
+
+inline void read_nested(
+  iobuf_parser& in, model::compression& el, size_t const bytes_left_limit) {
+    serde::read_enum(in, el, bytes_left_limit);
+}
```

Comment:
perhaps we should move these out of the reflection namespace?

Reply:
removed since I switched the PR back to adl